src/System.Diagnostics.TraceSource/src/System/Diagnostics/DefaultTraceListener.cs

```diff
@@ -179,13 +152,23 @@ private void WriteLine(string message, bool useLogFile)
 
         private void Write(string message, bool useLogFile)
         {
-            if (NeedIndent) 
+            if (message == null)
+            {
+                Write(string.Empty, useLogFile);
```

Comment:
This recursively calls the same method. Would it be better to just do `if (message == null) message = string.Empty;` ?